File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/Array_List/DoublePointer/List/deleteDuplicates_83.java
package algo.tzashinorpu.SpecialSubject.Array_List.DoublePointer.List;

import algo.tzashinorpu.SpecialSubject.ListNode;

public class deleteDuplicates_83 {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode dummyHead = new ListNode();
        dummyHead.next = head;
        ListNode r = head;
        while (r != null) {
            while (r != null && head.val == r.val) r = r.next;
            head.next = r;
            head = r;
        }
        return dummyHead.next;
    }
}

File: Clion/ZeroTrac/array_list/deleteDuplicates_83.cpp
//
// Created by TzashiNorpu on 10/18/2022.
//
#include "array_list.h"

using namespace Array_List;

ListNode *Solution::deleteDuplicates(ListNode *head) {
  ListNode *dummy = new ListNode();
  dummy->next = head;
  if (!head) return dummy->next;
  ListNode *slow = head, *fast = head->next;
  while (fast) {
    if (fast->val == slow->val)
      fast = fast->next, slow->next = fast;
    else fast = fast->next, slow = slow->next;
  }
  return dummy->next;
}